src/main/java/emissary/util/search/ByteMatcher.java

```diff
@@ -13,14 +13,30 @@
     public static final int NOTFOUND = -1;
 
 
+    public ByteMatcher(String data) {
+        this(data.getBytes());
```

Comment:
It would be better to specify a charset, or ensure one is provided by the client

Reply:
Added the ability to specify a charset by name